### PR TITLE
Improve the jerry port api documents.

### PR DESCRIPTION
**Pull request by lygstate**
Notable changes:
  - Remove the comments in port impl, that's easily getting to in-consistence
  - Sync the jerryscript-port.h and 05.PORT-API.md
  - Fixes the invalid comment in port codes

JerryScript-DCO-1.0-Signed-off-by: Yonggang Luo luoyonggang@gmail.com

### CODE REVIEW

**Comment by lygstate** [comment]
@LaszloLango Any suggestion about this?

**Review by LaszloLango — APPROVED** [review]
LGTM

File: jerry-port/common/jerry-port-context.c [hunk 1]
```diff
@@ -27,13 +27,6 @@
  */
 static jerry_context_t *current_context_p = NULL;
 
-/**
```
Comment (zherczeg) [inline jerry-port/common/jerry-port-context.c:30]: Why these are removed?

**Review by zherczeg — COMMENTED** [review]
A lot of comments are removed. Why?

File: jerry-port/common/jerry-port-context.c (hunk 1 above)
Comment (lygstate) [reply jerry-port/common/jerry-port-context.c:30]: Remove the comments in port impl, that's easily getting to in-consistence

Those already documented in header, document twice create a lot of inconsistence, and not updated. So just leave a single copy of the comment

Comment (LaszloLango) [reply jerry-port/common/jerry-port-context.c:30]: I agree with @lygstate, these comments should be place in the header file only.

**Review by robertsipka — APPROVED** [review]
LGTM

**Pull request merged**